src/main/java/frc/robot/subsystems/ball/BallImpl.java

```diff
@@ -85,21 +86,22 @@ public void retract() {
     public void task() {
         if(this.shootEnabled) {
             this.shooter.runShooter();
-            this.feeder.enableHopper();
+            this.feeder.enable(FeederComponent.HOPPER);
             if(this.shooter.readyForBalls()) {
-                this.feeder.enableConveyor();
+                this.feeder.enable(FeederComponent.CONVEYOR);
             } else {
-                this.feeder.stopConveyor();
+                this.feeder.stop(FeederComponent.CONVEYOR);
             }
         } else {
             if(this.feeding == FeederStates.OFF){
```

Comment:
This if-else should probably also be a switch, for consistency with other places where you're handling enums for these internal states.

Reply:
Yeah makes sense